keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,363 @@
+
+* applications leveraging different isolated networks exposed through different
+  interfaces
+* applications leveraging performance-oriented interfaces (e.g. `AF_XDP`,
+  `memif`, `SR-IOV`)
+* applications requiring support for protocols not yet supported by Kubernetes
+
+Here we propose a solution that enables multiple network interfaces to be
+defined in the `Pod` specification to provide a standard and clear mechanism to
+implement these more complex configurations.
+
+## Motivation
+
+We want to have a common API allowing us to define a catalog of different
+networks in the Kubernetes cluster. It would allow attaching a pod to one or
+several networks via a given type of interface depending on its connectivity
+or performance needs.
+
+### Goals
+
+* Define user stories and requirements for the Multi-Network effort in Kubernetes.
+* Introduce API object to kubernetes describing networks Pod can attach to.
+* Evolve current Kubernetes networking model to support multiple networks,
+defining the new model in a backwards compatible way.
+* Integrate this API with the existing core objects.
+* Define “reference implementation” that can be used in Kubernetes CI.
+* Define conformance tests.
+
+
+### Non-Goals
+
+Define the CNI implementation.
+
+## Proposal
+
+### Personas
+
+#### Cluster operator
+
+This role will be responsible for defining and managing PodNetworks (see below)
+that properly describe the infrastructure available for the cluster, namespace
+and workloads. This persona can define which users can “attach” to a specific
+PodNetwork.
+
+#### Application developer
+
+**Application developer** is the consumer of PodNetwork via referencing them in
+their workloads. Application developers usually will not create or remove the
+PodNetwork on their own.
+
+### Terminology
+
+* **PodNetwork** - name of the API object that will represent a network in
+  Kubernetes cluster
+* **Cluster Default PodNetwork** - This is the PodNetwork attached to the Pod
+  when no additional networking configuration is provided in Pod spec.
+* **Primary PodNetwork** - This is the PodNetwork inside the Pod which interface
+  is used for the default gateway.
+
+
+### User Stories
+
+All user stories represent the type of use cases the multi-networking API should
+be able to support. References to technologies or exact products does not
+indicate that this API will directly support them. User stories can be
+overlapping in some areas. We want to document the concrete use cases from an
+end-user standpoint -- they may (and should!) boil down to a few common
+primitives listed in the below requirements section.
+
+
+#### Story #1
+As a Cloud Native Network Function (CNF) vendor my workloads deals with 2 types
+of network traffic: control plane and dataplane. For regulatory compliance I have
+to isolate these 2 traffic types. To achieve this I require 2 separate network
+interfaces inside my Pod. The underlying implementation (outside the Pod) has to
```

Comment:
TIL that network policies are complaint with PCI DSS https://cloud.google.com/blog/products/containers-kubernetes/new-pci-dss-policy-bundle , honestly didn't know it